### DISCUSSION TITLE
How expensive is bind group creation, and is there any way to update a bind group?

### DISCUSSION THREAD

**Discussion by spencerkohan** [gfx-rs/wgpu#3138] [Q&A]
I'm trying to understand the best practices and performance implications of bind groups bette

Coming from Vulkan, you would generally want to cache descriptor sets when possible, and you might want to update a descriptor set rather than allocating a new descriptor set when possible.

With wgpu, I don't see a way to update a bind group, is that correct?

Also what is the performance impact of creating bind groups?  For instance, if I create new bind groups every frame, would this be considered an anti-pattern?

**Comment by cwfitzgerald** [comment]
> With wgpu, I don't see a way to update a bind group, is that correct?

Yes

> Also what is the performance impact of creating bind groups? For instance, if I create new bind groups every frame, would this be considered an anti-pattern?

Depends on how many and how big. Making 5-10 bind groups a frame, meh whatever. Making 500 I would start looking at carefully.
